beacon_node/beacon_chain/src/light_client_server_cache.rs

```diff
@@ -312,6 +370,96 @@ impl<T: BeaconChainTypes> LightClientServerCache<T> {
         self.latest_optimistic_update.read().clone()
     }
+
+    pub fn get_sync_committee_branch(
+        &self,
+        store: &BeaconStore<T>,
+        block_root: &Hash256,
+    ) -> Result<Option<FixedVector<Hash256, CurrentSyncCommitteeProofLen>>, BeaconChainError> {
+        let column = DBColumn::SyncCommitteeBranch;
+
+        if let Some(bytes) = store
+            .hot_db
+            .get_bytes(column.into(), &block_root.as_ssz_bytes())?
+        {
+            // TODO unwrap
+            let sync_committee_branch: FixedVector<Hash256, CurrentSyncCommitteeProofLen> =
+                FixedVector::from_ssz_bytes(&bytes).unwrap();
+
+            return Ok(Some(sync_committee_branch));
+        }
+
+        Ok(None)
+    }
+
+    pub fn get_sync_committee(
+        &self,
+        store: &BeaconStore<T>,
+        sync_committee_period: u64,
+    ) -> Result<Option<SyncCommittee<T::EthSpec>>, BeaconChainError> {
+        let column = DBColumn::SyncCommittee;
+
+        if let Some(bytes) = store
+            .hot_db
+            .get_bytes(column.into(), &sync_committee_period.as_ssz_bytes())?
+        {
+            let sync_committee: SyncCommittee<T::EthSpec> =
+                SyncCommittee::from_ssz_bytes(&bytes).unwrap();
+            return Ok(Some(sync_committee));
+        }
+
+        Ok(None)
+    }
+
+    #[allow(clippy::type_complexity)]
+    pub fn get_light_client_bootstrap(
+        &self,
+        store: &BeaconStore<T>,
+        block_root: &Hash256,
+        finalized_period: u64,
+        chain_spec: &ChainSpec,
+    ) -> Result<Option<(LightClientBootstrap<T::EthSpec>, ForkName)>, BeaconChainError> {
+        if let Some(block) = store.get_full_block(block_root)? {
+            let (state_root, slot) = (block.state_root(), block.slot());
+
+            let fork_name = chain_spec.fork_name_at_slot::<T::EthSpec>(slot);
+
+            let sync_committee_period = block
+                .slot()
+                .epoch(T::EthSpec::slots_per_epoch())
+                .sync_committee_period(chain_spec)?;
+
+            let current_sync_committee_branch =
+                self.get_sync_committee_branch(store, block_root)?.unwrap();
+            // TODO unwrap
+            // db.current_sync_committee_branch.get(block_root)
+
+            let current_sync_committee = if sync_committee_period <= finalized_period {
+                // fetch the header current sync committee from the db, since its not cached
+                // TODO unwrap
+                Arc::new(
+                    self.get_sync_committee(store, sync_committee_period)?
+                        .unwrap(),
+                )
+            } else {
+                // were at the current finalized period, so this data should be easily retrievable from the currently cached state
+                // TODO unwrap
+                let state = store.get_state(&state_root, Some(slot))?.unwrap();
+                state.current_sync_committee()?.clone()
```

Comment:
Even tho the state may be in the store it's not guaranteed. If we hit this branch we are potentially doing a lot of work: state load, apply HDiffs, block replay. I would delete this path and ensure that we always have a sync_committee cached ready for any slot.

EDIT: get bootstrap is meant to only support finalized block roots. So no need to support a case where 
`sync_committee_period > finalized_period`